Sources/Web3Auth/SessionManagement+Extension.swift

```diff
@@ -0,0 +1,134 @@
+        return string
+    }
+
+    func encParamsBufToHex(encParamsHex: ECIES) -> ECIES {
+        return .init(iv: encParamsHex.iv.web3.hexData?.web3.hexString ?? "", ephemPublicKey: encParamsHex.ephemPublicKey.web3.hexData?.toHexString() ?? "", ciphertext: encParamsHex.ciphertext.web3.hexData?.toHexString() ?? "", mac: encParamsHex.mac.web3.hexData?.web3.hexString ?? "")
+    }
+
+    private func encParamsHexToBuf(encParamsHex: String) throws -> ECIES {
+        let data = encParamsHex.data(using: .utf8) ?? Data()
+        var arr = Array(repeating: "", count: 4)
+        do {
+            let dict = try? JSONSerialization.jsonObject(with: data) as? [String: String]
+            dict?.forEach { key, value in
+                if key == "iv" {
+                    arr[0] = value
+                } else if key == "ephemPublicKey" {
+                    arr[1] = value
+                } else if key == "ciphertext" {
+                    arr[2] = value
+                } else if key == "mac" {
+                    arr[3] = value
+                }
+            }
+            return ECIES(iv: arr[0], ephemPublicKey: arr[1], ciphertext: arr[2], mac: arr[3])
+        } catch let error {
+            throw Web3AuthError.runtimeError(error.localizedDescription)
+        }
+    }
+
+    func encrypt(publicKey: String, msg: String, opts: ECIES?) throws -> ECIES {
+        guard let ephemPrivateKey = SECP256K1.generatePrivateKey(), let ephemPublicKey = SECP256K1.privateToPublic(privateKey: ephemPrivateKey)
+        else {
+            throw Web3AuthError.runtimeError("Private key generation failed")
+        }
+        let ephermalPublicKey = publicKey.strip04Prefix()
+        let ephermalPublicKeyBytes = ephermalPublicKey.hexa
+        var ephermOne = ephermalPublicKeyBytes.prefix(32)
+        var ephermTwo = ephermalPublicKeyBytes.suffix(32)
+        ephermOne.reverse(); ephermTwo.reverse()
+        ephermOne.append(contentsOf: ephermTwo)
+        let ephemPubKey = secp256k1_pubkey.init(data: array32toTuple(Array(ephermOne)))
+        guard
+            // Calculate g^a^b, i.e., Shared Key
+            //  let data = inprivateKey
+            let sharedSecret = SECP256K1.ecdh(pubKey: ephemPubKey, privateKey: ephemPrivateKey)
+        else {
+            throw Web3AuthError.unknownError
+        }
+
+        let sharedSecretData = sharedSecret.data
+        let sharedSecretPrefix = Array(tupleToArray(sharedSecretData).prefix(32))
+        let reversedSharedSecret = sharedSecretPrefix.uint8Reverse()
+        let hash = SHA2(variant: .sha512).calculate(for: Array(reversedSharedSecret))
+        let iv: [UInt8] = (opts?.iv ?? SECP256K1.randomBytes(length: 16)?.toHexString())?.hexa ?? []
+        let encryptionKey = Array(hash.prefix(32))
+        let macKey = Array(hash.suffix(32))
+        do {
+            // AES-CBCblock-256
+            let aes = try AES(key: encryptionKey, blockMode: CBC(iv: iv), padding: .pkcs7)
```

Comment:
look into CryptoKit AES functions

Reply:
this code is the same as torus-utils-swift right?